### PR TITLE
Static viz path for the updated trend chart

### PR DESCRIPTION
**Pull request by alxnddr**
Related to https://github.com/metabase/metabase/pull/37038

### Description

Creates a static visualization rendering path with a data contract similar to the application's data visualization contract to ensure maximum reuse of settings computation and actual viz rendering code.

In the scope of ECharts migration we did a similar change [here](https://github.com/metabase/metabase/pull/34820/commits/3f1ad8d6f1d9b5298a25f5533ab684ae835b7743#diff-98aa068e0239e92cdc383ef4a069b7563a7aed3e842509f83c24b9561b3260af) which we will rebase on top of the current PR change once it is merged. In the referred commit the rendering method is called `isomorphic` which was not the most precise name and `javascript_visualization` makes the code a bit more obvious.

### How to verify

1. Create a Trend chart
2. Send a test dashboard subscription with it
3. Ensure it renders the HTML placeholder

### Demo

Screenshot 2024-01-08 at 10 49 12 PM

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by alxnddr** [comment]
@markbastian can you please help with BE CI checks?

File: frontend/src/metabase/static-viz/components/StaticChart/StaticChart.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const Placeholder = ({ text }: { text: string }) => {
+  return <div style={{ width: "300px", height: "200px" }}>{text}</div>;
+};
+
+export interface StaticChartProps {
+  rawSeries: RawSeries;
+  dashcardSettings: VisualizationSettings;
+  renderingContext: RenderingContext;
+}
```
Comment (kulyk) [inline frontend/src/metabase/static-viz/components/StaticChart/StaticChart.tsx:11]: I belive every static chart component would need that type, right? Would it make sense to pull it up? I was thinking of `metabase/static-viz/types` or maybe `metabase/visualization/types`?

Comment (kulyk) [reply frontend/src/metabase/static-viz/components/StaticChart/StaticChart.tsx:11]: Also, it seems like we're missing a proper type for `RenderingContext`?

Comment (alxnddr) [reply frontend/src/metabase/static-viz/components/StaticChart/StaticChart.tsx]: Good catch, thanks! Fixed in the latest commit

**Review by kulyk — APPROVED** [review]
FE LGTM 👍

**Review by markbastian — APPROVED** [review]
I fixed a breaking BE issue I found and did a small amount of cleanup so returned `:style` is a keyword. 🚢 🇮🇹

**Pull request merged**